subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasFailedTestDeterminer.kt

```diff
@@ -0,0 +1,111 @@
+package com.avito.instrumentation.report
+
+import com.avito.report.model.SimpleRunTest
+import org.funktionale.tries.Try
+
+interface HasFailedTestDeterminer {
+
+    fun determine(
+        runResult: Try<List<SimpleRunTest>>
+    ): Result
+
+    sealed class Result {
+
+        open fun count(): Int = 0
```

Comment:
Default implementation can break LSP

Reply:
What is LSP?

Reply:
I agree. It's maybe not about LSP.

DetermineError(error).count() -> 0, this confused me, that count() makes no sense in some children